components/proxy/src/main/java/com/hotels/styx/startup/ProxyServerSetUp.java

```diff
@@ -66,18 +66,20 @@ private static void initialisePlugins(Iterable<NamedPlugin> plugins) {
         if (exceptions > 0) {
             throw new RuntimeException(format("%s plugins failed to start", exceptions));
         }
+
+        configStore.set("plugins", plugins);
```

Comment:
Ideally this would move inside the loop:

```
        for (NamedPlugin plugin : plugins) {
            try {
                plugin.styxStarting();
                configStore.set("plugins", pluginsInitialisedSoFar);    // <----
            } catch (Exception e) {
                exceptions++;
                LOG.error("Error starting plugin '{}'", plugin.name(), e);
            }
        }
```

This would give a more instantaneous feedback about the plugins loaded so far. The end user could monitor the progress of plugins being initialised. Maybe something to consider in future.

However the current implementation is fine for now.

Reply:
I had the same thought, but for now the current implementation is simpler.